server/utils.py

```diff
@@ -50,6 +56,12 @@ def get_auth_response(auth_status):
         return make_response(jsonify({'message': 'Error: Unexpected authentication status'}), 500)
 
 
+def auth_token_validation(auth_token):
+    auth_status = get_auth_status(auth_token)
+    if auth_status != 'valid':
+        return get_auth_response(auth_status)
```

Comment:
This doesn't behave quite the way you're using it. This return only returns to the caller, it doesn't short-circuit a flask response. (i.e., it either returns a flask response or None, but you don't do anything with the response) Take a look at view decorators: http://flask.pocoo.org/docs/0.10/patterns/viewdecorators/